<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>867</width>
    <height>880</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>550</width>
          <height>300</height>
         </size>
        </property>
        <property name="title">
         <string>Sourcemeter Configuration</string>
        </property>
        <widget class="QLabel" name="label">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>30</y>
           <width>400</width>
           <height>250</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="pixmap">
          <pixmap>device_diagram.png</pixmap>
         </property>
         <property name="scaledContents">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QWidget" name="verticalLayoutWidget_2">
         <property name="geometry">
          <rect>
           <x>430</x>
           <y>60</y>
           <width>111</width>
           <height>191</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="smu_1_sn">
            <property name="text">
             <string>SN: -</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="smu_1_mode">
            <item>
             <property name="text">
              <string>Sweep</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Constant</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="smu_1_supply">
            <item>
             <property name="text">
              <string>Voltage</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Current</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="smu_2_sn">
            <property name="text">
             <string>SN: -</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="smu_2_mode">
            <property name="currentText">
             <string>Sweep</string>
            </property>
            <item>
             <property name="text">
              <string>Sweep</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Constant</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="smu_2_supply">
            <item>
             <property name="text">
              <string>Voltage</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Current</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="frame">
        <property name="frameShape">
         <enum>QFrame::Shape::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Shadow::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="QGroupBox" name="sourcemeter_config_group">
           <property name="title">
            <string>Sourcemeter 1 Configuration - Sweep Voltage</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="sweep_step_1">
                <property name="suffix">
                 <string> V</string>
                </property>
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="minimum">
                 <double>0.001000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>0.010000000000000</double>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="sweep_start_label_1">
                <property name="text">
                 <string>Voltage Start</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignmentFlag::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLabel" name="sweep_step_label_1">
                <property name="text">
                 <string>Voltage Step</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignmentFlag::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QDoubleSpinBox" name="sweep_start_1">
                <property name="suffix">
                 <string> V</string>
                </property>
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="minimum">
                 <double>-1100.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>1100.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>-1.500000000000000</double>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QLabel" name="sweep_end_label_1">
                <property name="text">
                 <string>Voltage End</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignmentFlag::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QDoubleSpinBox" name="sweep_end_1">
                <property name="suffix">
                 <string> V</string>
                </property>
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="minimum">
                 <double>-1100.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>1100.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>1.500000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout">
              <item row="0" column="0">
               <widget class="QLabel" name="supply_label_1">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Voltage Supply</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="supply_1">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="suffix">
                 <string> V</string>
                </property>
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="minimum">
                 <double>-1100.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>1100.000000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>0.000000000000000</double>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="compliance_label_1">
                <property name="text">
                 <string>Current Compliance</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="compliance_1">
                <property name="suffix">
                 <string> A</string>
                </property>
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="minimum">
                 <double>-5.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>5.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>0.020000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="sourcemeter_config_group_2">
           <property name="title">
            <string>Sourcemeter 2 Configuration - Constant Current</string>
           </property>
           <property name="test" stdset="0">
            <string>A, F, D</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="0">
               <widget class="QLabel" name="sweep_start_label_2">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Voltage Start</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignmentFlag::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLabel" name="sweep_step_label_2">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Voltage Step</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignmentFlag::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QLabel" name="sweep_end_label_2">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Voltage End</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignmentFlag::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="sweep_step_2">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="suffix">
                 <string> V</string>
                </property>
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="minimum">
                 <double>0.001000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>0.010000000000000</double>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QDoubleSpinBox" name="sweep_end_2">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="suffix">
                 <string> V</string>
                </property>
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="minimum">
                 <double>-1100.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>1100.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>1.500000000000000</double>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QDoubleSpinBox" name="sweep_start_2">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="suffix">
                 <string> V</string>
                </property>
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="minimum">
                 <double>-1100.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>1100.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>-1.500000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_4">
              <item row="0" column="0">
               <widget class="QLabel" name="supply_label_2">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="text">
                 <string>Current Supply</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="compliance_label_2">
                <property name="text">
                 <string>Voltage Compliance</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="supply_2">
                <property name="suffix">
                 <string> A</string>
                </property>
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="minimum">
                 <double>-5.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>5.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>0.020000000000000</double>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="compliance_2">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="suffix">
                 <string> V</string>
                </property>
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="minimum">
                 <double>-1100.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>1100.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>-1.500000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_6">
           <property name="title">
            <string>Quick Measurement Parameters</string>
           </property>
           <layout class="QVBoxLayout" name="_2">
            <item>
             <layout class="QGridLayout" name="gridLayout_5">
              <item row="0" column="0">
               <widget class="QLabel" name="label_17">
                <property name="text">
                 <string>Sweep Step Override</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="quick_measurement_sweep_step">
                <property name="suffix">
                 <string> V</string>
                </property>
                <property name="decimals">
                 <number>3</number>
                </property>
                <property name="minimum">
                 <double>0.001000000000000</double>
                </property>
                <property name="maximum">
                 <double>999.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.001000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>0.200000000000000</double>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_16">
                <property name="text">
                 <string>Pause Between Measurements</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="quick_measurement_pause">
                <property name="suffix">
                 <string> s</string>
                </property>
                <property name="decimals">
                 <number>2</number>
                </property>
                <property name="maximum">
                 <double>9999.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>0.010000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="quick_measurement_run">
              <property name="text">
               <string>Run Quick Measurement</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_7">
           <property name="title">
            <string>Measurement Parameters</string>
           </property>
           <layout class="QVBoxLayout" name="_3">
            <item>
             <layout class="QGridLayout" name="gridLayout_6">
              <item row="0" column="0">
               <widget class="QLabel" name="label_18">
                <property name="text">
                 <string>Number of Measurements</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_19">
                <property name="text">
                 <string>Pause Between Measurements</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="quick_measurement_pause_2">
                <property name="suffix">
                 <string> s</string>
                </property>
                <property name="decimals">
                 <number>2</number>
                </property>
                <property name="maximum">
                 <double>9999.000000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
                </property>
                <property name="value">
                 <double>0.010000000000000</double>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QSpinBox" name="quick_measurement_sweep_step_2">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="value">
                 <number>2</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="quick_measurement_run_2">
              <property name="text">
               <string>Run Measurements</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="graph_container"/>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
